source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -158,6 +156,12 @@ void ProxyFilter::addHostAddressToFilterState(
     return;
   }
 
+  if (!address) {
+    ENVOY_STREAM_LOG(warn, "Cannot add address to filter state: invalid address",
+                     *decoder_callbacks_);
+    return;
+  }
```

Comment:
I don't think this should happen? Or does this happen if there is no address? If so I would make this debug as this could happen at high volume.

Reply:
I did see it happen for v6 tests. I dont have v6 configured locally so that might be triggering it. Also, not sure why the v6 integration test was failing (v4 works fine).

Reply:
OK can you debug a bit why it's happening? Is this case when there is no address? If so that is a normal case so I would just drop to debug.

Reply:
I see log 
`[debug][forward_proxy] [source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc:365] DNS refresh rate reset for host 'localhost:44099', (failure) refresh rate 60000 ms
`
So i think given the severity of log(`debug`), it could happen. I will change the log severity to `debug`.  Integration test fails for Windows. CI logs does not have much detail:

https://dev.azure.com/cncf/4684fb3d-0389-4e0b-8251-221942316e06/_apis/build/builds/88602/logs/113